src/read_log.py

```diff
@@ -0,0 +1,24 @@
+"""
+Read Log File
+
+By: Norma Chang
+"""
+import json
+import time
+
+def read_file():
+  """
+  Read the log file containing the cycle data.
+  """
+  string_arr = []
+  array_dict = []
+
+  log_file = open('log.txt', 'r')
+  log = log_file.read().strip().split('\n')
+
+  for line in log:
+    string_arr.append(line)
+
+  for string in string_arr:
+    dictionary = json.loads(string)
+    array_dict.append(dictionary)
```

Comment:
Absolutely not how that should work. The log file **needs** to be in an array form. The whole point of doing this in JSON is so we don't have to write our own parser to interpret the data. All I need to do is open the file and load it as a JSON when I'm using the data.

Reply:
It is not possible to predict the end of a continuous log (there is a super hack way but we should be doing things properly). The second function gives you an array of dictionaries. You would just need to call the function. It loads the json for you when you are creating a graph.